### PR TITLE
[KED-2969] - Fix the issue with showDiff when comparing with run without tracking data

### PR DESCRIPTION
**Pull request by rashidakanchwala**
## Description

There was a bug in showDiff wherein if one run had dataset that had no tracking data and another run had dataset which had some tracking data, it wouldn't show the common data between the two runs i.e. no data because there's nothing common. 
This issue has been fixed in this PR. 

## Checklist

- [ ] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added new entries to the `RELEASE.md` file
- [ ] Added tests to cover my changes

## Legal notice

- [x] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

- I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
- I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorised to submit this contribution on behalf of the original creator(s) or their licensees.
- I certify that the use of this contribution as authorised by the Apache 2.0 license does not violate the intellectual property rights of anyone else.

### CODE REVIEW

**Review by studioswong — CHANGES_REQUESTED** [review]
I've just tested this and the `showDiff` variable and it works well for both `true` and `false` conditions, thanks for pushing this fix!

Can you please add the related tests for both `showDiff = true` and `showDiff = false` so the expected behaviour for this variable will be clear in the codebase ( to prevent confusion down the line). Thank you.

File: package/tests/test_api/test_graphql.py [hunk 1]
```diff
@@ -87,6 +87,52 @@ def example_multiple_run_tracking_catalog(save_version, save_new_version):
...
+    # Note - filepath is assigned without using tmp_path as it fails on windows build.
+    # This is a temp soln and will be cleaned up in the future.
+    new_metrics_dataset = MetricsDataSet(
+        filepath="test.json",
+        version=Version(None, save_version),
+    )
+    new_metrics_dataset = MetricsDataSet(
+        filepath="test.json",
+        version=Version(None, save_new_version),
+    )
```
Comment (limdauto) [inline package/tests/test_api/test_graphql.py:125]: Why are there 2 objects with the same name here?

File: package/tests/test_api/test_graphql.py [hunk 2]
```diff
@@ -202,6 +248,104 @@ def test_graphql_run_tracking_data_query_show_diff_false(
                     ),
                 )
             ]
+    def test_graphql_run_tracking_data_show_diff_true_one_empty_run(
```
Comment (limdauto) [inline package/tests/test_api/test_graphql.py:251]: You can use https://docs.pytest.org/en/6.2.x/parametrize.html to make these tests cleaner. They are essentially 4 variations of the same test, not 4 separate tests IMHO.

**Review by limdauto — COMMENTED** [review]
I think the test could be a bit cleaner (added a couple of comments) but logic looks good to me.

File: package/tests/test_api/test_graphql.py (hunk 1 above)
Comment (rashidakanchwala) [reply package/tests/test_api/test_graphql.py:125]: I am trying to create two runs for the same metrics -- using two versions save_version, and save_new_version.

File: package/tests/test_api/test_graphql.py (hunk 2 above)
Comment (rashidakanchwala) [reply package/tests/test_api/test_graphql.py:251]: Thanks, this looks good. Will parametrize the below!!

**Review by limdauto — APPROVED** [review]
LGTM!

**Review by studioswong — APPROVED** [review]
Thanks for adding the tests!

**Pull request merged**